src/utils/displayTo.ts

```diff
@@ -1,33 +1,92 @@
 import { css, ThemedCssFunction } from './styled';
-import { ScreenSize, Breakpoints } from '../common';
+import { ScreenSize, Breakpoints, ScreenSizeList } from '../common';
 import { breakpoints } from '../themes';
 
-function isBreakpoint(breakpoints: Breakpoints, name: keyof Breakpoints) {
-  const value = breakpoints && breakpoints[name];
-  return typeof value === 'number' && value > 0;
+export function getMediaQueries(breakpoints: Breakpoints) {
+  return ScreenSizeList.map(x => ({ screen: x, breakpoints: getScreenSizeBreakpoints(x, breakpoints) })).reduce(
+    (acc, next) => (next.breakpoints ? { ...acc, [next.screen]: formatQuery(next.breakpoints) } : acc),
+    {},
+  );
 }
 
-export function getMediaQueries(breakpoints: Breakpoints) {
-  if (!isBreakpoint(breakpoints, 'medium') || !isBreakpoint(breakpoints, 'large')) {
-    throw new Error('Invaild breakpoints');
+// create max-width media query including screen size
+export function displayUpTo(screen: ScreenSize) {
+  const screenBreakpoints = getScreenSizeBreakpoints(screen, breakpoints);
+  if (!screenBreakpoints) {
+    throw new Error('Invalid screen size');
   }
+  const query = formatQuery({ next: screenBreakpoints.next });
+  return formatMedia(query);
+}
 
-  return {
-    small: `(max-width: ${breakpoints.medium - 1}px)`,
-    smallAndMedium: `(max-width: ${breakpoints.large - 1}px)`,
-    medium: `(min-width: ${breakpoints.medium}px) and (max-width: ${breakpoints.large - 1}px)`,
-    mediumAndLarge: `(min-width: ${breakpoints.medium}px)`,
-    large: `(min-width: ${breakpoints.large}px)`,
-  };
+// create min-width media query including screen size
+export function displayFrom(screen: ScreenSize) {
+  const screenBreakpoints = getScreenSizeBreakpoints(screen, breakpoints);
+  if (!screenBreakpoints) {
+    throw new Error('Invalid screen size');
+  }
+  const query = formatQuery({ prev: screenBreakpoints.prev });
+  return formatMedia(query);
 }
 
 export function displayTo(screen: ScreenSize | string): ThemedCssFunction<any> {
-  const query = getMediaQueries(breakpoints)[screen] || screen;
-  const result: ThemedCssFunction<any> = (strings: any, ...interpolations: Array<any>) =>
+  const screenBreakpoints = getScreenSizeBreakpoints(screen as ScreenSize, breakpoints);
+  const query = screenBreakpoints ? formatQuery(screenBreakpoints) : screen;
+  return formatMedia(query);
+}
+
+// create media query based on previous and next breakpoints
+export function formatQuery(breakpoints: { prev?: number; next?: number }) {
```

Comment:
Btw, if it's a public API, the `breakpoints` arg should be `{ min: number, max: number }` to be more readable as well as developer-friendly.
Just compare:
`displayTo(formatQuery({ prev: 300, next: 600}))` vs `displayTo(formatQuery({ min: 300, max: 600}))`

And last but not least you should validate whether `min < max` if both breakpoints provided.

Reply:
```suggestion
export function formatMediaQuery(breakpoints: { prev?: number; next?: number }) {
```

Reply:
Updated

Reply:
Added formatQuery example